### ISSUE TITLE
`pip install -e .` on framework sources fails

### ISSUE THREAD

**Issue by riccardomurri** [hpcugent/easybuild-framework#1708]
When I run `pip install .` on a fresh "easybuild-framework" checkout, I get this error::

        $ pip install -e .
        Obtaining file:///home/rmurri/w/easybuild/framework
            Complete output from command python setup.py egg_info:
            ERROR: Running EasyBuild from /home/rmurri/w/easybuild/framework does not work (Python packaging weirdness)...

            ----------------------------------------
        Command "python setup.py egg_info" failed with error code 1 in /home/rmurri/w/easybuild/framework

If I run `python setup.py develop` from the same directory, everything works fine.

Since the two commands are (supposedly) functionally equivalent, they
should both work the same way.

It may related to this issue already discussed on the mailing list:
<https://lists.ugent.be/wws/arc/easybuild/2016-03/msg00050.html>

**Comment by boegel** [comment]
@riccardomurri This is caused by the following check in `easybuild/__init__.py`:

```python
# check whether EasyBuild is being run from a directory that contains easybuild/__init__.py;
# that doesn't work (fails with import errors), due to weirdness to Python packaging/setuptools/namespaces
if __path__[0] == 'easybuild':
    sys.stderr.write("ERROR: Running EasyBuild from %s does not work (Python packaging weirdness)...\n" % os.getcwd())
    sys.exit(1)
```

I put this check in place after running into weird `import` issues when the `easybuild` module was being imported from the cloned git repo (either framework or easyblocks).

It seems that `pip` is something importing `easybuild`, thus triggering the check and failing.
I'd like to revisit whether the check is really needed though...
I can't seem to reproduce the problems now, maybe they were related to the use of `fixup_namespace_packages`, which I dropped last-minute before the EasyBuild v2.7.0 release (cfr. #1680).

However, I tried reproducing your problem, and couldn't after a quick try...
Which setuptools version are you using here (seems relevant)?

**Comment by boegel** [comment]
Should be fixed by #1730

**Comment by boegel** [comment]
@riccardomurri care to verify whether this problem is fixed on top of the branch I used for #1730?

**Comment by riccardomurri** [comment]
The fix in PR #1730 works for me.

**Comment by boegel** [comment]
Great, thanks for testing @riccardomurri!
